executor/common_linux.h

```diff
@@ -2417,17 +2416,24 @@ static unsigned long fs_image_segment_check(unsigned long size, unsigned long ns
 		size = IMAGE_MAX_SIZE;
 	return size;
 }
-#endif
 
-#if SYZ_EXECUTOR || __NR_syz_read_part_table
-// syz_read_part_table(size intptr, nsegs len[segments], segments ptr[in, array[fs_image_segment]])
-static long syz_read_part_table(volatile unsigned long size, volatile unsigned long nsegs, volatile long segments)
+// Setup the loop device needed for mounting a filesystem image. Takes care of
+// creating and initializing the underlying file backing the loop device and
+// returns the fds to the file and device.
+// Returns 0 on success, -1 otherwise.
+static int setup_loop_device(long unsigned size, long unsigned nsegs, struct fs_image_segment* segs, const char* memfd_name, const char* loopname, int* _memfd, int* _loopfd)
 {
-	char loopname[64], linkname[64];
-	int loopfd, err = 0, res = -1;
-	unsigned long i, j;
-	size = fs_image_segment_check(size, nsegs, segments);
-	int memfd = syscall(sys_memfd_create, "syz_read_part_table", 0);
+	int err = 0;
+	int loopfd;
+	int memfd;
+
+	if (!loopname || !memfd_name || !_memfd || !_loopfd) {
```

Comment:
We don't do this level of defensive programming.
We have tests and this clutters C reproducers.
memfd_name you pass as non-NULL just a few lines below. The same for loopname. Checking every variable for NULL every other line is just not worth it.
And if you are thinking about fuzzer-provided pointers, you can't defend against them. It won't give you NULL, it will give you 0x555555555555555555 ;)